features/device_key/source/DeviceKey.h~HEAD

```diff
@@ -0,0 +1,142 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_DEVICEKEY_H
```

Comment:
@geky Your view is clear and you presented it several times.
Review of this PR should be done in terms of code quality and not feature definitions. There are good reasons to keep two different APIs for storage but this is not the place to discuss this.
Will appreciate your help on code quality review

Reply:
I hope you will find the review itself is constructive.

However this is the correct place to note why a PR should not be merged, otherwise the PR will be merged.

Reply:
The Head file is part of the dependent DeviceKey PR and should be handled there. @yossi2le are you aware of this?

Reply:
Maybe it was accidentally added while rebasing. Will check.

Reply:
No rush, you'll need to rebase when dependencies go in anyways